README.md

```diff
@@ -124,3 +128,71 @@ interpreting them with HIBA.
 libhiba provides convenience functions to parse and extract HIBA extensions
 directly from a certificate, but it does not verify the certificate. It assumes
 the certificate was validated by a prior stage.
+
+### Example CA
+
+Once you have completed the
+[compilation instructions](https://github.com/google/hiba/blob/main/INSTALL.md#compilation)
+you will have a `hiba-gen` and `hiba-ca.sh` in your hiba directory. You can now
+create CA keys, grants, identities, grant eligibility, and sign host and user
+certificates with these extensions. The following assumes you are in your hiba
+directory.
+
+1. Choose a directory for the CA structure.
+  * `$ export dest=/tmp/hiba-ca`
+1. Generate a CA keypair.
+  * ```
+$ ./hiba-ca.sh -c -d "$dest" -- -N ""
+== Generating CA keys in /tmp/hiba-ca/ca
+== Done ==
+```
+1. Generate a test user keypair (alternatively you can copy existing keys to
+`$dest/users/`).
+  * ```
+$ ./hiba-ca.sh -c -d "$dest" -u -I testuser -- -N ""
+== Generating user keys in /tmp/hiba-ca/users/testuser
+== Done ==
+Identity created: Private: /tmp/hiba-ca/users/testuser
+                  Public:  /tmp/hiba-ca/users/testuser.pub
+```
+1. Copy the host public key from the OpenSSH server to the `$dest/hosts`
+directory as {devicename}.pub.
+  * ```
+$ cat $dest/hosts/test.pub
+
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC0xYoE3JGnRAfRaqjckXE4maij08NzmXcaSCZO0/vbE7Hi6OgaSDGzNoQyY10X1rwOziRhrpE2iHpX+NLbw4P7UpNIUowjodqqBi/pB/h22YqwxXBbxMq6SiVASLx31NpbNFJyeqzafjeXs5/+54mYwVMBivHrXt4vnqojgMVWMDm9fgBEoEeSQYR7cEIrFR4wPJDO70qWny2Uehj7uxKAm5sg4hb6g9akDRArsvvHDDcrvPFzJq0qqCmMhmjkYjeYExBs2Wi5noMGgdJpFVAW6VzxLkFWq4G1v8fQzgQZGROApzssK0rddIkZn4rJ+UIlq3nXGGAVFOQWCKhKx5Emv36QQIANx8Q+z1WFdqkOENLJRxpTvOP6Vj9U91YQZX4L4L3lZeGnxmYy3zmRClX3F70vhy0QD3kOLdX2qssj1l39ej2FUWxA2p65AruATWxKkXYGWqx42cM08Mujh9NBqnejanZifLhu4lCuN57oi5ydQbQGc2kqGmY9tTiWy9k=
+```
+1. Create an identity for prod, with a domain of example.com.
+  * ```
+$ ./hiba-gen -i -f "$dest/policy/identities/prod" domain example.com
+```
+1. Create a grant called shell with a matching domain.
+  * ```
+$ ./hiba-gen -f "$dest/policy/grants/shell" domain example.com
+```
+1. Allow testuser to request this grant (shell).
+  * ```
+$ ./hiba-ca.sh -d "$dest" -p -I testuser -H shell
+== Done ==
+User testuser is now eligible for [shell]
+```
+1. Sign the OpenSSH server public key into a certificate. -I is the basename of
+   the public key in $dest/hosts.
+  * ```
+$ ./hiba-ca.sh -s -d "$dest" -h -I test -n test -H prod -- -P ""
```

Comment:
The hostname is automatically passed as principal, -n is for adding more principals, and it is not required here.